lib/json/ld/decoder.ex

```diff
@@ -5,113 +5,113 @@ defmodule JSON.LD.Decoder do
   use RDF.Serialization.Decoder
 
   import JSON.LD.{NodeIdentifierMap, Utils}
-  alias JSON.LD.NodeIdentifierMap
-  alias RDF.{Dataset, Graph, NS}
+
+  alias JSON.LD.{NodeIdentifierMap, Options}
+  alias RDF.{BlankNode, Dataset, Graph, IRI, Literal, NS, Statement, XSD}
 
   @impl RDF.Serialization.Decoder
+  @spec decode(String.t(), keyword) :: {:ok, Dataset.t() | Graph.t()} | {:error, any}
   def decode(content, opts \\ []) do
     with {:ok, json_ld_object} <- parse_json(content),
-         dataset = to_rdf(json_ld_object, opts) do
+         dataset <- to_rdf(json_ld_object, opts) do
```

Comment:
It came from Credo recommendations: 
`with` doesn't end with a <- clause, move the non-pattern <- clauses inside the body of the `with`

In case you do want to revert, I will do it, no problem.

Reply:
I would prefer moving the clause into the body. Doing an unneeded pattern matching is definitely worse.

Reply:
Ok, done.